domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputHasIntegerPartEqualToRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,277 @@
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.app.model.Fraction
+import org.oppia.android.app.model.InteractionObject
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.cast
+import kotlin.test.fail
+
+/** Tests for [FractionInputHasIntegerPartEqualToRuleClassifierProvider]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class FractionInputHasIntegerPartEqualToRuleClassifierProviderTest {
+  @Inject
+  internal lateinit var fractionInputHasIntegerPartEqualToRuleClassifier:
+    FractionInputHasIntegerPartEqualToRuleClassifierProvider
+
+  private val inputHasIntegerPartEqualToRuleClassifier by lazy {
+    fractionInputHasIntegerPartEqualToRuleClassifier.createRuleClassifier()
+  }
+
+  private val FRACTION_1_OVER_2 =
+    createFraction(isNegative = false, numerator = 1, denominator = 2)
+  private val FRACTION_4_OVER_2 =
+    createFraction(isNegative = false, numerator = 4, denominator = 2)
+  private val FRACTION_5_OVER_2 =
+    createFraction(isNegative = false, numerator = 5, denominator = 2)
+  private val FRACTION_2_OVER_1 =
+    createFraction(isNegative = false, numerator = 2, denominator = 1)
+  private val NEGATIVE_FRACTION_1_OVER_2 =
+    createFraction(isNegative = true, numerator = 1, denominator = 3)
+  private val NEGATIVE_FRACTION_4_OVER_2 =
+    createFraction(isNegative = true, numerator = 1, denominator = 2)
+  private val MIXED_NUMBER_123_1_OVER_2 =
+    createMixedNumber(isNegative = false, wholeNumber = 123, numerator = 1, denominator = 2)
+  private val NEGATIVE_MIXED_NUMBER_123_1_OVER_2 =
+    createMixedNumber(isNegative = true, wholeNumber = 123, numerator = 1, denominator = 2)
+  private val STRING_VALUE = createString(value = "test")
+  private val WHOLE_NUMBER_0 = createNonNegativeInt(value = 0)
+  private val WHOLE_NUMBER_1 = createNonNegativeInt(value = 1)
+  private val WHOLE_NUMBER_2 = createNonNegativeInt(value = 2)
+  private val WHOLE_NUMBER_123 = createNonNegativeInt(value = 123)
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testAnswer1Over2_input0_HasIntegerPartEqual() {
+    val inputs = mapOf("x" to WHOLE_NUMBER_0)
+
+    val matches = inputHasIntegerPartEqualToRuleClassifier.matches(
+      answer = FRACTION_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testAnswer5Over2_input2_HasIntegerPartEqual() {
+    val inputs = mapOf("x" to WHOLE_NUMBER_2)
+
```

Comment:
This test is failing, I am a bit confused  here because when I tested it with ```WHOLE_NUMBER_0```, it passed.

Reply:
I think it is something like our discussion [here](https://github.com/oppia/oppia-android/pull/1932#issuecomment-704135077).  
We don't look at the actual value of the answer ( which is 2(1/2) in this case ), instead, we look at the fraction as is (5/2) and 5/2 doesn't have an integar part equal to 2.  
But it has an integar part equal to 0.  
Does this help?

Reply:
Thanks

Reply:
@MohamedMedhat1998 Please resolve this comment.